src/Models/Solution.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace SolutionsService.Models
 {
-    public class Solution
+    public abstract class Solution
```

Comment:
I'm not sure we need to apply inheritance here. Are different solutions significantly different enough between them, or are empty fields acceptable? Inheritance adds a lot of complexity and handling, where we could otherwise choose to never return empty fields instead.

Reply:
Not applying inheritance implies we need to do a LOT of validation on request to ensure that fields that need to be empty are in fact empty. This would actually lead to much more complexity and inefficiency later down the road.

Reply:
I'm not sure it's more complex, since all of the validation on the fields that are allowed to be filled will always be the same and not different, making the validation on the empty fields as easy as a `if(SolutionType=x)`, which could easily be one method used on all of the fields being validated, since all it cares about in this case is if they're empty or not. Or it could be as easy as an attribute on the field (https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/attributes/)

Having to cast and check and handle inheritance on a DB level seems like it'll add more complexity than that. Though if we think in the future dozens more types of solutions will be made this'll be true.

Reply:
On the DB level, things are handled entirely by Entity Framework which means none of the complexity has to be handled by us